sjb/syntax.md

```diff
@@ -0,0 +1,211 @@
+# `sjb` YAML Syntax Reference
+
+A job configuration using `sjb` will take one of two forms: a stand-alone job
+or one that inherits from another. A standalone job can declare options; a job
+with a parent can extend or override any option from the parent.
+
+Parent-child relationships can nest -- to generate a job, the topmost parent
+configuration is loaded, then iteratively extended and overridden by children
+configurations until the bottommost configuration has been evaluated.
+
+## `parameters`
+
+`parameters` is an optional field which adds parameters to the Jenkins job. The
+field contains a list of parameter definitions. The parameter definition syntax
+is:
+
+```yaml
+parameters:
+  - name: ""          # environment variable name
+    description: ""   # human-readable description
+    default_value: "" # default value [optional]
```

Comment:
They're all Bash variables in the end so I'm not super inclined to do that, as they will all be strings in the end and at least if they start as strings it's easier to understand what the variables will contain in the script where you use them.